src/nbl/asset/utils/IShaderCompiler.cpp

```diff
@@ -83,11 +82,7 @@ inline core::smart_refctd_ptr<ICPUShader> nbl::asset::IShaderCompiler::compileTo
         auto compressedSpirvBuffer = core::make_smart_refctd_ptr<ICPUBuffer>(propsSize + destLen);
         memcpy(compressedSpirvBuffer->getPointer(), compressedSpirv.data(), compressedSpirvBuffer->getSize());
 
-        entry.dependencies = std::move(dependencies);
-        entry.spirv = std::move(compressedSpirvBuffer);
-        entry.uncompressedSize = spirvBuffer->getSize();
-
-        options.writeCache->insert(std::move(entry));
+        options.writeCache->insert(std::move(SEntry(entry, std::move(dependencies), std::move(compressedSpirvBuffer), spirvBuffer->getContentHash(), spirvBuffer->getSize())));
```

Comment:
pro C++ tip (I don't actually want you to change anything), there's an `emplace` intead of `insert` for this (constructs an element in-place